action/sealedenvelope.go

```diff
@@ -7,19 +7,39 @@ import (
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 	"github.com/pkg/errors"
 
+	"github.com/iotexproject/iotex-core/config"
 	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
 )
 
 // SealedEnvelope is a signed action envelope.
 type SealedEnvelope struct {
 	Envelope
+	encoding      uint32
+	externChainID uint32
+	srcPubkey     crypto.PublicKey
+	signature     []byte
+}
 
-	srcPubkey crypto.PublicKey
-	signature []byte
+// RawHash returns the raw hash of embedded Envelope
+// this is the hash to be signed
+func (sealed *SealedEnvelope) RawHash() hash.Hash256 {
+	if IsRLP(iotextypes.Encoding(sealed.encoding)) {
+		if tx, err := actionToRLP(sealed.Action()); err == nil {
+			return rlpRawHash(tx, sealed.externChainID)
+		}
+		return hash.ZeroHash256
+	}
+	return sealed.Envelope.Hash()
```

Comment:
if we keep this function, Envelope.Hash, it will be confusing, because it is just the hash of the protobuf encoding format

Reply:
i think the logic is similar to what we discussed for signed hash?
```
if rlp()
   return RLP's raw hash
return sealed.Envelope.Hash(), which is native protobuf's raw hash
```

Reply:
Envelope.Hash is only used in Sign and Verify, in other word, Hash doesn't need to be an essential public API of Envelope.
Same for https://github.com/iotexproject/iotex-core/pull/2592/files#diff-fda8afdb7165ec30b8d2c40dfd1963def0ffeb117a3d9098167b8139c72527caR25, RawHash doesn't expose to external packages

Reply:
ok, right, RawHash() need not expose

Reply:
will have another PR to remove Envelope.Hash